lib/tracker2/modes2/decode_adsb.go

```diff
@@ -0,0 +1,530 @@
+package modes2
+
+import (
+	"bytes"
+	"errors"
+	"math"
+)
+
+var (
+	aisCharset = "@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_ !\"#$%&'()*+,-./0123456789:;<=>?"
+)
+
+// DecodeADSB is long for performance reasons, no need to copy memory around on stacks
+//
+//nolint:funlen
+func (a AvrFrame) DecodeADSB() (DFADSB, error) {
```

Comment:
Is it worth considering a `sync.Pool` of `DFADSB` types? Instead of creating a new one here, we could pull from the pool, and return the object to the pool at the end of its life. This way we would reduce memory allocations and possibly speed things up even more?

Reply:
Neverymind, just read your initial comment.

Reply:
The initial implementation does use it, I was trying out using the stack to have the data copied around.

it's fast, but not as fast as validating pointers apparently